packages/graph/hash_graph/lib/graph/src/store/postgres/query/statement/select.rs

```diff
@@ -374,36 +374,74 @@ mod tests {
-    //
-    //     let filter = Filter::Equal(
-    //         Some(FilterExpression::Path(EntityTypeQueryPath::Links(
-    //             LinkTypeQueryPath::Title,
-    //         ))),
-    //         Some(FilterExpression::Parameter(Parameter::Text(Cow::Borrowed(
-    //             "Friend Of",
-    //         )))),
-    //     );
-    //     compiler.add_filter(&filter);
-    //
-    //     test_compilation(
-    //         &compiler,
-    //         r#"
-    //         SELECT *
-    //         FROM "entity_types"
-    //         INNER JOIN "entity_type_link_type_references" AS
-    // "entity_type_link_type_references_0_0"           ON
-    // "entity_type_link_type_references_0_0"."source_entity_type_version_id" =
-    // "entity_types"."version_id"         INNER JOIN "link_types" AS "link_types_0_1"
-    //           ON "link_types_0_1"."version_id" =
-    // "entity_type_link_type_references_0_0"."target_link_type_version_id"         WHERE
-    // "link_types_0_1"."schema"->>'title' = $1         "#,
-    //         &[&"Friend Of"],
-    //     );
-    // }
+    #[test]
+    fn entity_type_by_referenced_link_types() {
+        let mut compiler = SelectCompiler::<EntityType>::with_asterisk();
+
+        let filter = Filter::Equal(
+            Some(FilterExpression::Path(EntityTypeQueryPath::Links(
+                Box::new(EntityTypeQueryPath::Links(Box::new(
+                    EntityTypeQueryPath::Title,
+                ))),
+            ))),
+            Some(FilterExpression::Parameter(Parameter::Text(Cow::Borrowed(
+                "Friend Of",
+            )))),
+        );
+        compiler.add_filter(&filter);
+
+        test_compilation(
+            &compiler,
+            r#"
+            SELECT *
+            FROM "entity_types" AS "entity_types_0_0_0"
+            INNER JOIN "entity_type_entity_type_references" AS "entity_type_entity_type_references_0_1_0"
+              ON "entity_type_entity_type_references_0_1_0"."source_entity_type_version_id" = "entity_types_0_0_0"."version_id"
+            INNER JOIN "entity_types" AS "entity_types_0_2_0"
+              ON "entity_types_0_2_0"."version_id" = "entity_type_entity_type_references_0_1_0"."target_entity_type_version_id"
+            INNER JOIN "entity_type_entity_type_references" AS "entity_type_entity_type_references_0_3_0"
+              ON "entity_type_entity_type_references_0_3_0"."source_entity_type_version_id" = "entity_types_0_2_0"."version_id"
+            INNER JOIN "entity_types" AS "entity_types_0_4_0"
+              ON "entity_types_0_4_0"."version_id" = "entity_type_entity_type_references_0_3_0"."target_entity_type_version_id"
+            WHERE jsonb_extract_path("entity_types_0_0_0"."schema", 'links', "entity_types_0_2_0"."schema"->>'$id') IS NOT NULL
+              AND jsonb_extract_path("entity_types_0_2_0"."schema", 'links', "entity_types_0_4_0"."schema"->>'$id') IS NOT NULL
+              AND "entity_types_0_4_0"."schema"->>'title' = $1
+            "#,
```

Comment:
Would it be useful for us to try to add an integration test for the above query? Or should we hold off to do that in a follow-up after dev/links?

Reply:
I'd hold off touching the integration test suite as this needs to be redone entirely. However, I'll add a test for the REST endpoints.

Reply:
I added a test in 5228618f4cb7883f9dc15298296a40db42c39283 which is checking if there is any entity link type that links to another entity link type (which should be none). Without this PR it would return the friend-of entity type.

Reply:
Thanks!